File: AE-ActiveExperiments/React-Redux-Workspace/02-react-redux-emp-app/src/service/EmployeeService.mjs
import axios from "axios";
class EmployeeService {
    baseUrl = `http://localhost:8085/Employee_app/employees/`;
    
    addEmployee(employee) {
        //console.log("inside service"+JSON.stringify(employee))
        return axios.post(this.baseUrl, employee);
    }
    deleteEmployeeById(empId) {
        //alert("inside delete employee "+empId)
        return axios.delete(this.baseUrl + empId);
    }
    getAllEmployees() {
        
        return axios.get(this.baseUrl);
    }
    updateEmployee(employee) {
        return axios.put(this.baseUrl+employee.employeeId, employee);
    
    }
    getAllDepartment() {
        return axios.get(this.baseUrl + 'getAllDepartments');
    }
    findEmployeeById(empId) {
        return axios.get(this.baseUrl +empId);
    }
}
export default EmployeeService;
// module.exports = {

//     EmployeeService

// }

File: AD-ActiveDevelopments/JEST_SpringRestEmployeeBackend_React_Redux_App2/src/main/java/com/priyanka/service/EmployeeService.java
package com.priyanka.service;

import java.util.List;

import com.priyanka.exceptions.NoSuchEmployeeException;
import com.priyanka.model.Employee;


public interface EmployeeService {

	public boolean addEmployee(Employee employee);
	public Employee findEmployeeById(int employeeId)throws NoSuchEmployeeException;
	public List<Employee> findAllEmployees();
	public Employee removeEmployeeById(int employeeId)throws NoSuchEmployeeException;
	public boolean updateEmployee(Employee employee);

}
